app/snapshot.go

```diff
@@ -234,6 +234,45 @@ func getCommonSnapshots(replicas []rest.Replica) ([]string, error) {
 	return snapshots, nil
 }
 
+func getAllSnapshots(replicas []rest.Replica) ([]string, error) {
+	first := true
+	finalSnapshotList := []string{}
+
+	for _, r := range replicas {
+		snapshots := []string{}
+		diffSnapshots := []string{}
+		if r.Mode != "RW" {
+			continue
+		}
+		replica, err := getReplica(r.Address)
+		if err != nil {
+			return nil, err
+		}
+		// Replica has just been started and haven't prepared the head
+		// file yet
+		if len(replica.Chain) == 0 {
+			break
+		}
+		for i, snap := range replica.Chain {
+			if i == 0 {
+				continue
+			}
+			if !replica.Disks[snap].Removed {
+				snapshots = append(snapshots, snap)
+			}
+		}
+
+		if !first {
+			diffSnapshots = util.Filter(snapshots, func(i string) bool {
+				return !util.Contains(finalSnapshotList, i)
+			})
+		}
+		first = false
+		finalSnapshotList = append(snapshots, diffSnapshots...)
```

Comment:
here, we might be trying to find common snapshots across all the replicas when all of them are in 'RW' state.
so, we should be using finalSnapshotList to append as well.

Reply:
ok.. we are getting all the snapshots, not just the common ones.
Why do we need them? and if so, its better to return the replica addresses also where it is currently available?

Reply:
but, we should still use finalSnapshotList to append and make some other changes.

Reply:
Will add replica address in a separate PR, fixed the above issue.